Add background scanning for devices

Knalltuete5000 opened the pull request:
Add a background routine to rescan in a specified interval for devices

By default the rescanning is triggered every 10 minutes. Please let me know if this interval is too short or too long or if the rescanning should be disabled by default. This can be done by setting the rescan interval to less than 1 seconds.

Fixes #130

k0ste commented:
I think, this will be good if add to logger msg "rescan for every `smartctlRescanInterval`"

k0ste commented:
BTW, tested this patch on hardware!

```python
ts=2023-08-08T09:36:14.000Z caller=main.go:135 level=info msg="Starting smartctl_exporter" version="(version=0.9.1, branch=master, revision=9d4d51b1495663f5150b8ff9d6c26e4180bd038c)"
ts=2023-08-08T09:36:14.000Z caller=main.go:136 level=info msg="Build context" build_context="(go=go1.20.7, platform=linux/amd64, user=k0ste@k0ste.ru, date=20230808-16:33:41, tags=unknown)"
ts=2023-08-08T09:36:14.000Z caller=main.go:142 level=info msg="No devices specified, trying to load them automatically"
ts=2023-08-08T09:36:14.015Z caller=main.go:115 level=info msg="Found device" name=/dev/sda
ts=2023-08-08T09:36:14.015Z caller=main.go:115 level=info msg="Found device" name=/dev/sdb
ts=2023-08-08T09:36:14.015Z caller=main.go:115 level=info msg="Found device" name=/dev/sdc
ts=2023-08-08T09:36:14.015Z caller=main.go:115 level=info msg="Found device" name=/dev/sdd
ts=2023-08-08T09:36:14.015Z caller=main.go:115 level=info msg="Found device" name=/dev/nvme0
ts=2023-08-08T09:36:14.015Z caller=main.go:157 level=info msg="Start background scann process"
ts=2023-08-08T09:36:14.285Z caller=tls_config.go:232 level=info msg="Listening on" address=[::]:9633
ts=2023-08-08T09:36:14.285Z caller=tls_config.go:235 level=info msg="TLS is disabled." http2=false address=[::]:9633
ts=2023-08-08T09:46:14.115Z caller=main.go:69 level=info msg="Rescanning for devices"
ts=2023-08-08T09:46:14.137Z caller=main.go:115 level=info msg="Found device" name=/dev/sda
ts=2023-08-08T09:46:14.138Z caller=main.go:115 level=info msg="Found device" name=/dev/sdb
ts=2023-08-08T09:46:14.138Z caller=main.go:115 level=info msg="Found device" name=/dev/sdc
ts=2023-08-08T09:46:14.138Z caller=main.go:115 level=info msg="Found device" name=/dev/sdd
ts=2023-08-08T09:46:14.138Z caller=main.go:115 level=info msg="Found device" name=/dev/nvme0
```

Knalltuete5000 commented:
Based on the feedback of @k0ste (thank you) I have added an additional log line.

In addition I have added a second condition, that a rescan is only executed if no static list of devices is configured. I hope this improves the quality of the feature.

k0ste commented on main.go:
```diff
@@ -135,13 +153,19 @@ func main() {
 		logger:  logger,
 	}
 
+	if *smartctlRescanInterval > 1*time.Second && len(*smartctlDevices) > 0 {
```
Should be greater or equal `>= 0`, because when devices are not defined is a zero len

Knalltuete5000 replied:
You're right. I think it should be `== 0`. So if any devices are configured via the the `smartctl.device` option, if they exist or not, the background task should not run. Am I right?

k0ste replied:
Seems right

k0ste commented:
@NiceGuyIT
